.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,28 @@
+name: SpringBootGroupProject CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+  merge_group:
+    types: [ checks_requested ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Retrieve Java Version
+        run: |
+          java_version=$(mvn help:evaluate "-Dexpression=maven.compiler.release" -q -DforceStdout | sed -e 's/^1\./1.0./')
+          echo "JAVA_VERSION=$java_version" >> $GITHUB_ENV
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: 'temurin'
+          cache: maven
```

Comment:
We need to call sonarcloud scanner and credentials from the yaml file. Otherwise it wont work. We can add it as a separate yaml file in later issue. 
In these projects  [Thunder](https://github.com/fungover/thunder/blob/main/.github/workflows/maven.yml) and
[Movies](https://github.com/Ebyrdeu/movie_jakarta/blob/master/.github/workflows/sonar.yml) you can see the different implementations. 
Maby the docs on [SonarScanner](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/sonarscanner-for-maven/) and [sonarcloud with github actions](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/github-actions-for-sonarcloud/) can help us when we want to get it done. I do think we can pass this.